arch/x86/adler32_avx512_vnni.c

```diff
@@ -7,66 +7,52 @@
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
 
+#ifdef X86_AVX512VNNI_ADLER32
+
 #include "../../zbuild.h"
 #include "../../adler32_p.h"
 #include "../../cpu_features.h"
 #include "../../fallback_builtins.h"
 #include <immintrin.h>
+#include "../../adler32_fold.h"
 #include "adler32_avx512_p.h"
+#include "adler32_avx2_p.h"
 
-#ifdef X86_AVX512VNNI_ADLER32
-Z_INTERNAL uint32_t adler32_avx512_vnni(uint32_t adler, const unsigned char *buf, size_t len) {
-    uint32_t sum2;
+Z_INTERNAL uint32_t adler32_avx512_vnni(uint32_t adler, const uint8_t *src, size_t len) {
+    uint32_t adler0, adler1;
+    adler1 = (adler >> 16) & 0xffff;
+    adler0 = adler & 0xffff; 
+
+    if (src == NULL) return 1L;
+    if (len == 0) return adler;
 
-    /* For impossibly tiny sizes, use the smaller width versions. We still need
-     * to check for compile time support for these but they are likely there */
-#ifdef X86_SSE41_ADLER32
+rem_peel:
     if (len < 32)
-        return adler32_sse41(adler, buf, len);
+#if defined(X86_SSSE3_ADLER32)
+        return adler32_ssse3(adler, src, len);
+#else
+        return adler32_len_16(adler0, src, len, adler1); 
 #endif
 
 #ifdef X86_AVX2_ADLER32
     if (len < 64)
-        return adler32_avx2(adler, buf, len);
+        return adler32_avx2(adler, src, len);
+#else
+        return adler32_len_16(adler0, src, len, adler1); 
```

Comment:
Identation might be off.

Reply:
Actually a few lines above, the `#if defined(X86_SSSE3_ADLER32)` goes under the if statement. Not sure if it is intended the same here.

Reply:
ah yeah, we actually want to check for the SSSE3 function here as well.  Fix incoming.